syntax = "proto3";
option py_generic_services = true;

import "google/protobuf/timestamp.proto";

service Eurotherm {
    // Terminate/stop server.
    rpc StopServer(StopRequest) returns (Empty) {}

    // Does nothing. Used to check sever health.
    rpc ServerHealthCheck(Empty) returns (Empty) {}

    // stream process values
    rpc StreamProcessValues(StreamProcessValuesRequest) returns (stream ProcessValues) {}
}

message Empty {}

message StopRequest {}

message StreamProcessValuesRequest {}

message ProcessValues {
    google.protobuf.Timestamp timestamp = 1;
    double processValue = 2;
    double measuredValue = 3;
    double workingSetpoint = 4;
    double workingOutput = 5;
}

